xbmc/utils/URIUtils.cpp

```diff
@@ -1019,12 +1020,9 @@ bool URIUtils::IsInternetStream(const CURL& url, bool bStrictCheck /* = false */
   if (url.IsProtocol("stack"))
     return IsInternetStream(CStackDirectory::GetFirstStackedFile(url.Get()));
 
-  // Special case these
-  //! @todo sftp special case has to be handled by vfs addon
-  if (url.IsProtocol("ftp") || url.IsProtocol("ftps")  ||
-      url.IsProtocol("dav") || url.IsProtocol("davs")  ||
-      url.IsProtocol("sftp"))
-    return bStrictCheck;
+  // Only consider "streamed" filesystems internet streams when being strict
+  if (bStrictCheck && IsStreamedFileSystem(url.Get())
+    return true;
```

Comment:
shouldn't this be:
```suggestion
    if (bStrictCheck)
      return IsStreamedFileSystem(url.Get());
```
?

Reply:
No, because in that case if you would pass something with http:// it would return false

Reply:
In that case I don't understand what strictCheck is for

Reply:
strictCheck means: "Do we consider quasi filesystems like webdav, ftp, sftp and ftps also streams?". Normally we don't.

Reply:
Ok, thanks.
The new code will always return true for http, on purpose or did you want to test for Dav in IsStreamedFilesystem?

Reply:
The old code also always returns true for http. This is also one of the reasons I refactored it a bit: to make it (hopefully) more obvious what the idea is...